src/edgepi/dac/edgepi_dac.py

```diff
@@ -84,16 +116,12 @@ def write_voltage(self, analog_out: DACChannel, voltage: float):
         Raises:
             `ValueError`: if voltage has more decimal places than DAC accuracy limit
         """
-        self.dac_ops.check_range(analog_out.value, 0, NUM_PINS-1)
-        self.dac_ops.check_range(voltage, 0, UPPER_LIMIT)
         dac_gain = CalibConst.DAC_GAIN_FACTOR.value if self.__get_gain_state() else 1
+        self.dac_ops.check_range(analog_out.value, 0, NUM_PINS)
+        self.dac_ops.check_range(voltage, 0, (UPPER_LIMIT*dac_gain)+0.001)
```

Comment:
What is this 0.001? Why check range twice? Not necessarily wrong, I just don't understand

Reply:
https://github.com/EdgePi-Cloud/edgepi-python-sdk/blob/56c69ca1f3f0fa74fa8e9e96094d08681158a4ac/src/edgepi/dac/dac_commands.py#L140-L144

check range condition changed for the max value. Therefore, to check the upper limit, 0.001V added. Now I realize it is better to change the UPPER_LIMIT constant to 5.0005 instead.

Reply:
df7eb18d92d9516015533bb2808e871da631df3a